### PR TITLE
Rename "Show all" in notifications API and web UI

### PR DESCRIPTION
**Pull request by dmarcoux**
It never showed all notifications. The "Show all" link and parameter show a maximum number of notifications (300 right now).

### CODE REVIEW

**Comment by dmarcoux** [comment]
[Rename "Show all" to "Show/Display up to X/Show more"](https://trello.com/c/TMaQQMJW/1808-rename-show-all-to-show-display-up-to-x-show-more)

File: src/api/public/apidocs-new/paths/my_notifications.yaml [hunk 1]
```diff
@@ -6,14 +6,17 @@ get:
   parameters:
     - in: path
       name: project
-      required: false
```
Comment (dmarcoux) [inline src/api/public/apidocs-new/paths/my_notifications.yaml:9]: By default, parameters aren't required, so it doesn't have to be documented.

**Review by saraycp — APPROVED** [review]

File: src/api/public/apidocs-new/paths/my_notifications.yaml [hunk 2]
```diff
@@ -6,14 +6,17 @@ get:
...
         type: string
     - in: path
       name: page
-      required: false
       schema:
         type: int
+    - in: path
+      name: show_maximum
+      schema:
+        type: int
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/my_notifications.yaml:18]:
```suggestion
        type: string
    - in: path
      name: show_maximum
      schema:
        type: string
```
According to [the documentation](https://swagger.io/docs/specification/data-models/data-types/#numbers), OpenAPI has two numeric types, `number` and `integer`. I would suggest to use `string`, as the current code doesn't strictly require an integer.

File: src/api/public/apidocs-new/paths/my_notifications.yaml [hunk 3]
```diff
@@ -6,14 +6,17 @@ get:
...
       name: project
-      required: false
       schema:
         type: string
     - in: path
       name: page
-      required: false
       schema:
         type: int
+    - in: path
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/my_notifications.yaml:15]:
```suggestion
    - in: query
```

I'm going to open another PR to change the other parameters too and add a few more that are missing...

**Review by saraycp — CHANGES_REQUESTED** [review]

**Review by eduardoj — APPROVED** [review]

**Review by saraycp — APPROVED** [review]

**Pull request merged**